### PR TITLE
fix(gleam): add scanner.c to parsers list & update lockfile

### PR DESCRIPTION
**Pull request by amaanq**
Supersedes #4462

### CODE REVIEW

**Comment by amaanq** [comment]
Gleam queries need an update, ugh. I'm going to take over as maintainer for Gleam then 😆

**Comment by despairblue** [comment]
I tried out your fork and I get errors inside the buffer now like this:

image

Can't copy them since I can't select them. I could try with tmux to select them. Not sure why they appear inside the buffer though.

**Comment by amaanq** [comment]
> I tried out your fork and I get errors inside the buffer now like this:
> 
> image
> 
> Can't copy them since I can't select them. I could try with tmux to select them. Not sure why they appear inside the buffer though.

working on it.. :)

**Comment by despairblue** [comment]
Sorry, thought I could help 😬 

I'll be patient. Thanks for doing this ❤️

**Comment by amaanq** [comment]
Hey hey sorry about the delay @despairblue, working on fixing the tests now but highlights are fine

**Comment by despairblue** [comment]
> Hey hey sorry about the delay @despairblue, working on fixing the tests now but highlights are fine

Tested it and it works. Thank you :)

**Comment by amaanq** [comment]
Hey @despairblue, I need some advice here. This test is failing since the grammar was changed heavily, and so the same items don't parse quite the same anymore. Here's an example file

image

And here's the actual output tree:
image

Is the "Ok" in 2:10 ( context: Ok(i) ) a type, enum, or constructor? or a builtin of one of the 3?

**Comment by rawhat** [comment]
The grammar defines this under `record_pattern` as `constructor_name`.  I think generally I'd consider this a constructor, although realistically all of those you mentioned could apply.

**Comment by amaanq** [comment]
> The grammar defines this under `record_pattern` as `constructor_name`. I think generally I'd consider this a constructor, although realistically all of those you mentioned could apply. I did ask in the gleam discord and got at least some confirmation of this, but could definitely be corrected here.

drop an invite? ;)

**Comment by rawhat** [comment]
https://gleam.run/community/ the link is here. I think the invite link is static, but I'm not sure.  This seemed better 😄

**Comment by amaanq** [comment]
@despairblue @rawhat try the newest changes now, I really dislike how the grammar and queries were written :disappointed:

**Comment by despairblue** [comment]
Wow, so much has happened since I went to sleep. 😄 

I'll have to walk the dog before I can try it out, look at the changes a bit more and follow along on discord. Will do today.

Thanks again, much appreciated

**Comment by despairblue** [comment]
I tried it locally and it highlights everything correctly. I just learning the language last month, so I don't have many opinions about how things should highlight yet 😄

**Comment by despairblue** [comment]
I do have `'nvim' terminated by signal SIGSEGV (Address boundary error)` a lot now. But that may be related to my other packages (since I synced all packages) or my system being out of date.

Let me update everything, reboot and see if it disappears.

**Comment by despairblue** [comment]
Yeah, my problem is unrelated to this, but I have to fix this somehow before I can continue testing. I opened an issue in the neovim repo. 🤷🏾

**Comment by despairblue** [comment]
Solved. Another plugin broke yesterday (nvim-ghost) and updating nvim-treesitter coincided with updating that plugin too.

This here works fabulously! ❤️

**Comment by amaanq** [comment]
Great to hear

**Review by ObserverOfTime — APPROVED** [review]

**Comment by despairblue** [comment]
Thank you @amaanq ❤️

File: tests/query/highlights/gleam/function.gleam [hunk 1]
```diff
@@ -60,68 +60,68 @@ fn list_of_two(my_value: a) -> List(a) {
...
 // <- punctuation.bracket
 
 pub external fn random_float() -> Float = "rand" "uniform"
 // <- type.qualifier
-//  ^ type.qualifier
-//           ^ keyword.function
-//              ^ function
+//  ^^^^^^^^ type.qualifier
+//           ^^ keyword.function
+//              ^^^^^^^^^^^^ function
```
Comment (theHamsta) [inline tests/query/highlights/gleam/function.gleam:106]: The checker crate and our test suite will just check the last caret. Should this become more fine-grained?

Comment (amaanq) [reply tests/query/highlights/gleam/function.gleam:106]: I'd say yes, it's nicer to cover in case of say builtin types vs regular type identifiers

**Pull request merged**